types/context.go

```diff
@@ -106,6 +112,11 @@ func (c Context) IsCheckTx() bool             { return c.checkTx }
 func (c Context) MinGasPrices() DecCoins      { return c.minGasPrice }
 func (c Context) EventManager() *EventManager { return c.eventManager }
 func (c Context) AppVersion() string          { return c.appVersion }
+func (c Context) WithCache(cache *Cache) Context {
```

Comment:
I don't see any usages of this function... We should remove and remove the tests too

Reply:
removed